heartbeat/SMBinclude

```diff
@@ -0,0 +1,493 @@
+            IFS=';' read -r -a array <<< "$OCF_RESKEY_extraopt_list"
+            for i in "${array[@]}";do
+                echo "        $i" >> $OCF_RESKEY_outfile;
+            done
+	fi
+        smbinclude_addinclude
+}
+
+smbinclude_delete() {
+        if [ -e $OCF_RESKEY_outfile ];then
+                rm -f $OCF_RESKEY_outfile 2>/dev/null
+                smbinclude_delinclude
+                exit $OCF_SUCCESS
+        fi
+}
+
+smbinclude_reloadconfig() {
+        smbcontrol smbd reload-config 2>/dev/null
+        if [ $? -eq 0 ];then
+                exit $OCF_SUCCESS
+        else
+                ocf_log err "Seems you have an error in your samba configuration"
+                exit $OCF_ERR_CONFIGURED
+        fi
+}
+
+smbinclude_checktmpmount() {
+
+        ISMOUNT=$(grep $OCF_RESKEY_confd /proc/mounts|grep -c tmpfs)
+        if [ $ISMOUNT -ne 1 ];then
+                ocf_log err "The directoy /etc/samba/conf.d need to be a tmpfs mount"
+                exit $OCF_ERR_CONFIGURED
+        fi
+}
+
+
+######################################################################
+
+smbinclude_usage() {
+        cat <<END
+usage: $0 {start|stop|status|monitor|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+	return $OCF_SUCCESS
+}
+
+smbinclude_start() {
+        smbinclude_create
+        smbinclude_reloadconfig
+}
+
+smbinclude_stop() {
+        smbcontrol smbd close-share $OCF_RESKEY_share > /dev/null 2>&1
+        smbinclude_delete
+        smbinclude_reloadconfig
+}
+
+smbinclude_monitor() {
+        RES=$(smbcontrol smbd ping > /dev/null 2>&1)
+        if [ $? -eq 0 ];then
+                if [ $(testparm -s 2>/dev/null| egrep -c \\[$OCF_RESKEY_share\\]) -eq 1 ];then
+                        return $OCF_SUCCESS
+                else
+                        return $OCF_NOT_RUNNING
+                fi
+        else
+                return $OCF_NOT_RUNNING
+        fi
+}
+
+smbinclude_state() {
+        smbinclude_checktmpmount
+        RES=$(smbcontrol smbd ping > /dev/null 2>&1)
+        if [ $? -eq 0 ];then
+                if [ $(testparm -s 2>/dev/null| egrep -c \\[$OCF_RESKEY_share\\]) -eq 1 ];then
+                        ocf_log info "Samba share $OCF_RESKEY_share is active"
+			return $OCF_SUCCESS
+                else
+                        ocf_log info "Samba share $OCF_RESKEY_share is not active"
+			return $OCF_NOT_RUNNING
+                fi
+        else
+                 ocf_log info "Samba share $OCF_RESKEY_share is not active"
+		 return $OCF_NOT_RUNNING
+        fi
+}
+
+
+
+case $__OCF_ACTION in
+meta-data)      meta_data
+                ;;
+usage|help)     smbinclude_usage
+                ;;
+esac
+
+
+case $__OCF_ACTION in
+start)          smbinclude_start
+                ;;
+stop)           smbinclude_stop
+                ;;
+status)         smbinclude_state
+                ;;
+monitor)        smbinclude_monitor
+                ;;
+validate-all)   ;;
```

Comment:
You should add a validate function to return `$OCF_SUCCESS` here to avoid $? being set by any of the above actions (or any future code that didnt expect it not to get set here).